spark/src/main/scala/org/apache/mahout/sparkbindings/drm/CheckpointedDrmSpark.scala

```diff
@@ -165,7 +168,14 @@ class CheckpointedDrmSpark[K: ClassTag](
       else if (classOf[Writable].isAssignableFrom(ktag.runtimeClass)) (x: K) => x.asInstanceOf[Writable]
       else throw new IllegalArgumentException("Do not know how to convert class tag %s to Writable.".format(ktag))
 
-    rdd.saveAsSequenceFile(path)
```

Comment:
Ok, so we can eliminate the runtime error by mapping each key/value to the appropriate `Writable` (not from within a function):

```
if (ktag.runtimeClass == classOf[Int]) {
      rddInput.toDrmRdd()

        .map( x =>(new IntWritable(x._1.asInstanceOf[Int]), new VectorWritable(x._2))).saveAsSequenceFile(path)

    } else if (ktag.runtimeClass == classOf[String]){
      rddInput.toDrmRdd()

        .map( x =>(new Text(x._1.asInstanceOf[String]), new VectorWritable(x._2))).saveAsSequenceFile(path)

    } else if (ktag.runtimeClass == classOf[Long]) {
      rddInput.toDrmRdd()

        .map( x =>(new LongWritable(x._1.asInstanceOf[Long]), new VectorWritable(x._2))).saveAsSequenceFile(path)

    } else throw new IllegalArgumentException("Do not know how to convert class tag %s to Writable.".format(ktag))```